internal/controllers/common/resource_definitions/resource_definitions.go

```diff
@@ -827,13 +837,33 @@ func NewKeystoreSecretForCR(cr *operatorv1beta1.Cryostat) *corev1.Secret {
 	}
 }
 
-func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) *corev1.ServiceAccount {
-	return &corev1.ServiceAccount{
+func NewServiceAccountForCR(cr *operatorv1beta1.Cryostat) (*corev1.ServiceAccount, error) {
+	OAuthRedirectReference := &oauthv1.OAuthRedirectReference{
+		Reference: oauthv1.RedirectReference{
+			Kind: "Route",
+			Name: cr.Name,
+		},
+	}
+
+	ref, err := json.Marshal(OAuthRedirectReference)
+	if err != nil {
+		return nil, err
+	}
+
+	sa := &corev1.ServiceAccount{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      cr.Name,
 			Namespace: cr.Namespace,
+			Labels: map[string]string{
+				"app": "cryostat",
+			},
+			Annotations: map[string]string{
+				"serviceaccounts.openshift.io/oauth-redirecturi.route":       "https://",
+				"serviceaccounts.openshift.io/oauth-redirectreference.route": string(ref),
+			},
 		},
 	}
```

Comment:
Although it's probably harmless, I think we should only set these annotations when running on OpenShift. A Kubernetes user might be confused why there are OpenShift annotations pointing to a non-existent Route. You can pass `r.IsOpenShift` when calling this method like we do for `NewDeploymentForCR`.